cloud/pkg/controllermanager/edgeapplication/overridemanager/resourcesoverrider.go

```diff
@@ -79,25 +79,53 @@ func buildResourcesPatchesWithPath(specContainersPath string, rawObj *unstructur
 		if container.(map[string]interface{})["name"] == resourcesOverrider.ContainerName {
 			resourcesPath := fmt.Sprintf("/%s/%d/resources", specContainersPath, index)
 			resourcesValue := resourcesOverrider.Value
-			var patch overrideOption
-			// Overridden with the current value regardless of whether the template value is empty
-			patch, _ = acquireOverride(resourcesPath, resourcesValue)
 
-			klog.V(4).Infof("[buildResourcesPatchesWithPath] containers patch info (%+v)", patch)
-			patches = append(patches, patch)
+			// Acquire selective override patches based on non-zero fields in resourcesValue
+			patchList, err := acquireOverride(resourcesPath, resourcesValue)
+			if err != nil {
+				return nil, err
+			}
+
+			klog.V(4).Infof("[buildResourcesPatchesWithPath] containers patch info (%+v)", patchList)
+			patches = append(patches, patchList...)
 		}
 	}
 	return patches, nil
 }
 
-// acquireOverrideOption for adding resources override.
-func acquireOverride(resourcesPath string, resourcesValue corev1.ResourceRequirements) (overrideOption, error) {
+// acquireOverrideOptions for adding selective resources override.
+func acquireOverride(resourcesPath string, resourcesValue corev1.ResourceRequirements) ([]overrideOption, error) {
 	if !strings.HasPrefix(resourcesPath, pathSplit) {
-		return overrideOption{}, fmt.Errorf("internal error: [acquireOverride] resourcesPath should start with / character")
+		return nil, fmt.Errorf("internal error: [acquireOverride] resourcesPath should start with / character")
```

Comment:
Use `errors.New()` maybe better than `fmt.Errorf`

Reply:
Thanks for the correction, I have made the modification.